mojo-py/mojoland/recipes/connoisseur.py

```diff
@@ -191,6 +191,8 @@ def _compare_nibbles(self, fresh, original, tolerance):
                 fline = lines_fresh[i]
                 if oline == lines_fresh[i]:
                     continue
+                if '`offset` column is not supported' in oline: # newer mojo supports offset column while old ones won't
```

Comment:
I don't understand this change. What case is this for? I think it's wrong to do this, because this could hide that something is not working, isn't it?. It is like v1.00 doesn't support offset and v1.10 does support offset, right? So _compare_nibbles() will always just either
1. take glm_v1.10_*.mojo, score it with currently installed h2o and compares results with results stored in time when it was created (with glm_v1.10_*_scores*.txt files). And it is being checked that the same mojo, this case v1.10, gives the same results regardless of the current h2o version vs. h2o version which was used in time when first results were generated and saved into glm_v1.10_*_scores*.txt files.

or
2. take glm_v1.00_*.mojo, score it with currently installed h2o and compares results with results stored in time when it was created (with glm_v1.00_*_scores*.txt files). And it is being checked that the same mojo, this case v1.00, gives the same results regardless of the current vs. previous h2o version.

So if offset is supported from v1.10, then I would expect some results like numbers in glm_v1.10_*_scores1.txt and some results like "java.lang.UnsupportedOperationException: `offset` column is not supported" in  glm_v1.00_*_scores1.txt

Reply:
I've run it without this change and got this result:

```
#----------------------------------------------------------
#  Tasting Glm Cars recipe
#----------------------------------------------------------

Tasting nibble parameters in /Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-data/mojos/glm/cars/v1.10/glm_v1.10_cars_parameters.txt... ok
Tasting nibble multiparams in /Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-data/mojos/glm/cars/v1.10/glm_v1.10_cars_multiparams.txt... ok
Tasting nibble scores0 in /Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-data/mojos/glm/cars/v1.10/glm_v1.10_cars_scores0.txt... ok
Tasting nibble scores1 in /Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-data/mojos/glm/cars/v1.10/glm_v1.10_cars_scores1.txt... ok
Tasting nibble scores2 in /Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-data/mojos/glm/cars/v1.10/glm_v1.10_cars_scores2.txt... ok
Tasting nibble parameters in /Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-data/mojos/glm/cars/v1.00/glm_v1.00_cars_parameters.txt... ok
Tasting nibble multiparams in /Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-data/mojos/glm/cars/v1.00/glm_v1.00_cars_multiparams.txt... ok
Tasting nibble scores0 in /Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-data/mojos/glm/cars/v1.00/glm_v1.00_cars_scores0.txt... ok
Tasting nibble scores1 in /Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-data/mojos/glm/cars/v1.00/glm_v1.00_cars_scores1.txt... fail

Difference in line 1:
Command: score0~dadda([8,360,175,3821,11,73], 0, [0, 0])
Original: java.lang.UnsupportedOperationException: `offset` column is not supported
Computed: [15.367033442452353, 0.0]

Traceback (most recent call last):
  File "in_mojo_veritas.py", line 40, in <module>
    connoisseur.degustate(recipe)
  File "/Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-py/mojoland/recipes/connoisseur.py", line 39, in degustate
    self._taste(recipe, version)
  File "/Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-py/mojoland/recipes/connoisseur.py", line 174, in _taste
    raise MojoUnstableError("Mismatch in nibble %s of mojo %s; new nibble saved in %s" %
mojoland.recipes.connoisseur.MojoUnstableError: Mismatch in nibble scores1 of mojo glm_v1.00_cars.mojo; new nibble saved in /Users/zuzanaolajcova/IdeaProjects/mojoland/temp/glm_v1.00_cars_scores1.txt
Closing connection _sid_b5ce at exit
H2O session _sid_b5ce closed.
make: *** [test-mojo-compatibility] Error 1
```

Even after incrementation of glm mojo to v1.10, when you score the v1.00 mojo, results should be always the same as they were - regardless of h2o version (whether with old h2o that supports only v1.00 or with new h2o that supports both v1.00 and v1.10). The contract which is to be tested is explained here: README.md. This will require change in https://github.com/h2oai/h2o-3/pull/5795 so that "old mojo gives old results".

Here is how Honza was fixing offset for XGB in the past: https://github.com/h2oai/mojoland/pull/9. 

So there shouldn't be this change in this file.

Reply:
@koniecsveta 

The confusing part is if you ran with new mojo and with offset, you are going to get a valid answer and not an error.  So, if you compare the new mojo result with old mojo result which basically contains an error being thrown, you will find it to be different.

I must be missing something.  I don't know how to make the new mojo throw the same error offset column not being supported in this case.  That is where the difficult is.

My fix is aiming to avoid this problem by if the old result is an error throw for offset column not being supported but the new mojo supported it, it is okay and we can allow it in this special case.  That is what my fix is about.  However, for numerical values, the new mojo and the old mojo results should be the same.

I think this is a special case because we are basically adding a new feature to fix the old error of offset column not being supported.  What do you think?